.gitignore

```diff
@@ -23,6 +23,7 @@ var/
 *.egg-info/
 .installed.cfg
 *.egg
+.vscode/
```

Comment:
As per my comment about `.python-version`, but the opposite I guess - is this a useful config file for others using this project and vs code, or should we be excluding them?

Reply:
If it is useful, might it be better to use a [.editorconfig](https://editorconfig.org/) (Eg [Django one](https://github.com/django/django/blob/master/.editorconfig), so it is editor agnostic?

Reply:
I'm not sure that it would be right to commit this as although I am sure there are some settings that are useful for everyone (e.g.enabling linting) there is probably others that could fall under personal preferences?

Reply:
I don't know if it does, but it would be neat if `.editorconfig` extended and overrides `.editor-specific-config`. Then we could check in the basics and any additional personal preferences would be taken care of in the local repo - with the local editor specific config file added to `.git/info/exclude` (or `.gitignore` if preferred - though again if I'd argue that's more for per project things (e.g. build artefacts) than individual preferences/tooling (e.g. `.idea/`)). Though it's probably easier to just say follow PEP8, run black, etc and let everyone take care of it however they say best - with CI saying 'nope' if it hasn't been done.

Reply:
How do `.editorconfig` and `pyproject.toml` interact?! Some projects like `black` use`pyproject.toml` for settings, but I don't know if that's standard or not?

This is all quite confusing. I suggest we invent our own file format to settle this once and for all :)